execution/executor-service/src/remote_cordinator_client.rs

```diff
@@ -29,22 +34,57 @@ impl RemoteCoordinatorClient {
         let result_tx =
             controller.create_outbound_channel(coordinator_address, execute_result_type);
 
+        let state_view_client =
+            RemoteStateViewClient::new(shard_id, controller, coordinator_address);
+
         Self {
+            state_view_client: Arc::new(state_view_client),
             command_rx,
             result_tx,
         }
     }
+
+    // Extract all the state keys from the execute block command. It is possible that there are duplicate state keys.
+    // We are not de-duplicating them here to avoid the overhead of deduplication. The state view server will deduplicate
+    // the state keys.
+    fn extract_state_keys(command: &ExecuteBlockCommand) -> Vec<StateKey> {
+        command
+            .sub_blocks
+            .sub_block_iter()
+            .flat_map(|sub_block| {
+                sub_block
+                    .transactions
+                    .par_iter()
```

Comment:
I think its okay here since the global thread pool is not doing anything else in the remote executor setup, but still will put a TODO here to re-evaluate this.